### PR TITLE
Refactor sensitivity time plot

### LINKED ISSUE #204: `sensitivityTimeProfiles` performance
Calling `sensitivityTimeProfiles` is slow, have to profile and try to optimize the performance.

### LINKED ISSUE #228: `sensitivityTimeProfiles` color palette
image

The current pallete is too pale, I am having hard times focusing on the lines. I would like to try out the esqlabs-palette.

### LINKED ISSUE #618: SA Time profile: `DefaultPlotConfiguration` properties that must be supported
List all properties of `DefaultPlotConfiguration`, that will be supported by the SA time profile plot.

### LINKED ISSUE #624: Function signature for `sensitivityTimeProfiles
E.g.:

```
sensitivityTimeProfiles(
  sensitivityCalculation,
  outputPaths = NULL,
  parameterPaths = NULL,
DefaultPlotConfiguration
)
```

### PR DESCRIPTION
**Pull request by rengelke**
- Refactored `sensitivityTimeProfiles` with new signature: closes #624, closes #618,  closes #204 

@param xAxisScale Character string, either "log" (logarithmic scale) or "lin"
(linear scale), to set the x-axis scale. Default is "lin".
@param yAxisScale Character string, either "log" or "lin", sets the y-axis
scale similarly to `xAxisScale`. Default is "log".
@param defaultPlotConfiguration An object of class `DefaultPlotConfiguration`
used to customize plot aesthetics. Plot-specific settings provided directly
to the function, such as `xAxisScale`, will take precedence over any
modifications in `defaultPlotConfiguration`. If not provided, default
settings are applied.

- Resolve #228 by improving color palettes, scaling and legend

### CODE REVIEW

File: tests/testthat/_snaps/sensitivity-time-profiles/multiple-output-path-profiles.svg
Comment (PavelBal) [inline tests/testthat/_snaps/sensitivity-time-profiles/multiple-output-path-profiles.svg:1]: The y-scale doesn't look right

- 0 cannot be shown on the log-axis
- Distance between the tics in log scale is expected to be equal (how it was in the previous implementation=)

**Review by PavelBal — CHANGES_REQUESTED** [review]
And also please add some snapshot tests for linear axis scale

File: R/sensivitity-time-profiles.R [hunk 1]
```diff
@@ -87,92 +149,195 @@ sensitivityTimeProfiles <- function(sensitivityCalculation,
...
+      if (is.null(plotConfiguration$linesColor)) {
+        plot <- plot +
+          colorspace::scale_color_continuous_diverging(
+            palette = "Berlin",
+            mid = log10(1),
+            transform = "log",
+            breaks = cBreaks
+          )
+      } else {
+        plot <- plot +
```
Comment (Felixmil) [inline R/sensivitity-time-profiles.R:305]: I would maybe use a predefined diverging palette like Berlin available here: https://colorspace.r-forge.r-project.org/articles/hcl_palettes.html#diverging-palettes

Comment (rengelke) [reply R/sensivitity-time-profiles.R:305]: We have the same taste:

 ```
plot +
          colorspace::scale_color_continuous_diverging(
            palette = "Berlin",
            mid = log10(1),
            transform = "log",
            breaks = cBreaks
          )
```

Comment (Felixmil) [reply R/sensivitity-time-profiles.R:305]: Oh, I missed that. Great !

**Review by Felixmil — APPROVED** [review]

**Review by PavelBal — APPROVED** [review]
I didn't review the code, the plots look good.

**Review by Laura-Villain — COMMENTED** [review]
Good for me :)

**Pull request merged**